src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -733,4 +735,11 @@ public static void onCommandRegister(CommandDispatcher<CommandSource> dispatcher
         RegisterCommandsEvent event = new RegisterCommandsEvent(dispatcher, environment);
         MinecraftForge.EVENT_BUS.post(event);
     }
+
+    public static ItemStack onFindAmmo(PlayerEntity playerEntity, ItemStack shootable, Predicate<ItemStack> ammoPredicate)
+    {
+        PlayerFindAmmoEvent event = new PlayerFindAmmoEvent(playerEntity, shootable, ammoPredicate);
+        MinecraftForge.EVENT_BUS.post(event);
```

Comment:
You aren't doing any checks for if it is cancelled here.

Reply:
That's because the cancellation doesn't change the behaviour in the factory.
The cancellation is for other mods to know if they should be continuing expected behaviour or if they should ignore the event if a stack has been provided by another mod.

I wasn't sure to use the result or cancellation system here, and haven't gotten a response from my usual group of suspects that has yielded a better option/answer.

Reply:
This is invalid since cancellation is now gone.